instance/models/openedx_instance.py

```diff
@@ -483,19 +480,17 @@ def _write_metadata_to_consul(self, configurations):
         :return: A pair (version, changed) with the current version number and
                  a bool to indicate whether the information was updated.
         """
-        agent = ConsulAgent(prefix=self.consul_prefix)
-        version_updated = False
-
-        version_number = agent.get('version') or 0
-        for key, value in configurations.items():
-            index, stored_value = agent.get(key, index=True)
-            cas = index if stored_value is not None else 0
-            agent.put(key, value, cas=cas)
-
-            if not version_updated and value != stored_value:
-                version_updated = True
-                version_number += 1
-                agent.put('version', version_number)
+        with ConsulAgent(prefix=self.consul_prefix) as agent:
+            version_updated = False
+            version_number = agent.get('version') or 0
+            for key, value in configurations.items():
+                index, stored_value = agent.get(key, index=True)
+                cas = index if stored_value is not None else 0
```

Comment:
what is this **cas** guarding against? While I would understand getting the index when the value is first read, then transformed and eventually written back iff the index did not change, I don't get what is the value here.

Reply:
Ok, I get it now :)